src/debugger/handler.rs

```diff
@@ -7,154 +8,205 @@ use dap::responses::{
 use dap::types::{Breakpoint, Capabilities, Source, StackFrame, StoppedEventReason, Thread};
 use tracing::trace;
 
-use crate::debugger::ServerResponse;
+use crate::CairoDebugger;
 
-pub fn handle_request(request: &Request) -> ServerResponse {
-    match &request.command {
-        // We have not yet decided if we want to support these.
-        Command::BreakpointLocations(_)
-        | Command::Cancel(_)
-        | Command::Completions(_)
-        | Command::DataBreakpointInfo(_)
-        | Command::Disassemble(_)
-        | Command::Disconnect(_)
-        | Command::Goto(_)
-        | Command::ExceptionInfo(_)
-        | Command::GotoTargets(_)
-        | Command::LoadedSources
-        | Command::Modules(_)
-        | Command::ReadMemory(_)
-        | Command::RestartFrame(_)
-        | Command::SetDataBreakpoints(_)
-        | Command::Restart(_)
-        | Command::SetExceptionBreakpoints(_)
-        | Command::TerminateThreads(_)
-        | Command::Terminate(_)
-        | Command::StepInTargets(_)
-        | Command::SetVariable(_)
-        | Command::SetInstructionBreakpoints(_)
-        | Command::SetExpression(_)
-        | Command::WriteMemory(_) => {
-            // If we receive these with current capabilities, it is the client's fault.
-            let msg = format!("Received an unsupported request: {request:?}");
-            ServerResponse::Error(msg)
-        }
+pub enum HandleResult {
+    Handled,
+}
 
-        // These may be supported after the MVP.
-        // Nonetheless, if we receive these with current capabilities,
-        // it is the client's fault.
-        Command::ReverseContinue(_) => {
-            ServerResponse::Error("Step back is not yet supported".into())
-        }
-        Command::StepBack(_) => ServerResponse::Error("Step back is not yet supported".into()),
-        Command::SetFunctionBreakpoints(_) => {
-            ServerResponse::Error("Set function breakpoints is not yet supported".into())
-        }
+impl CairoDebugger {
+    pub(crate) fn handle_request(&self, request: Request) -> Result<HandleResult> {
```

Comment:
```suggestion
    pub fn handle_request(&self, request: Request) -> Result<HandleResult> {
```
This module is not visible from lib.rs anyways